File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    first = [1,2,3,4,5]*2000
    second = [2,1,2,3,2,4,2,5]*1250
    thrid = [3,3,1,1,2,2,4,4,5,5]*1000
    
    #answers의 길이로 다 잘라보자
    first=first[:len(answers)]
    second=second[:len(answers)]
    third=thrid[:len(answers)]
    
    #맞는 개수를 찾아야해
    first_correct=0
    second_correct=0
    third_correct=0
    for i in range(len(answers)):
        if answers[i]==first[i]:
            first_correct+=1
        if answers[i]==second[i]:
            second_correct+=1
        if answers[i]==third[i]:
            third_correct+=1
    
    max_= max(first_correct, second_correct, third_correct)
    
    if first_correct == max_ : answer.append(1)
    if second_correct == max_ : answer.append(2)
    if third_correct == max_ : answer.append(3)
        
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> scores={0,0,0,0};
    
    //1번 12345
    vector<int> first = {1,2,3,4,5}; 
    int first_score = 0;
    
    //2번 21232425
    vector<int> second = {2,1,2,3,2,4,2,5}; 
    int second_score = 0;
    
    //3번 3311224455
    vector<int> third = {3,3,1,1,2,2,4,4,5,5};
    int third_score = 0;
    
    for(int i=0;i<answers.size();i++){
        if(answers[i]==first[i%first.size()]){
            scores[1]++;
        }
        if(answers[i]==second[i%second.size()]){
            scores[2]++;
        }
        if(answers[i]==third[i%third.size()]){
            scores[3]++;
        }
    }
    
    int T= max_element(scores.begin(), scores.end())-scores.begin();
    //가장 큰 값은 일단 scores[T]
    for (int i=0;i<scores.size();i++){
        if(scores[i] == scores[T]){
            answer.push_back(i);
        }
    }
    
    
        
    
    
    
    return answer;
}